openedx/core/lib/gating/api.py

```diff
@@ -299,3 +304,112 @@ def get_gated_content(course, user):
                 {'id': user.id}
             )
         ]
+
+
+def compute_is_prereq_met(content_id, user_id, recalc_on_unmet=False):
+    """
+    Returns true if the prequiste has been met for a given milestone.
+    Will recalculate the subsection grade if specified and prereq unmet
+
+    Arguments:
+        content_id (BlockUsageLocator): BlockUsageLocator for the content
+        user_id: The id of the user
+        recalc_on_unmet: Recalculate the grade if prereq has not yet been met
+
+    Returns:
+        tuple: True|False,
+        prereq_meta_info = { 'url': prereq_url|None, 'display_name': prereq_name|None}
+    """
+    course_id = content_id.course_key
+
+    # if unfullfilled milestones exist it means prereq has not been met
+    unfulfilled_milestones = milestones_helpers.get_course_content_milestones(
+        course_id,
+        content_id,
+        'requires',
+        user_id
+    )
+    prereq_met = not unfulfilled_milestones
+    prereq_meta_info = {'url': None, 'display_name': None}
+
+    if prereq_met or not recalc_on_unmet:
+        return prereq_met, prereq_meta_info
+
+    milestone = unfulfilled_milestones[0]
+    student = User.objects.get(id=user_id)
+    store = modulestore()
+
+    with store.bulk_operations(course_id):
+        subsection_usage_key = UsageKey.from_string(_get_gating_block_id(milestone))
+        subsection_structure = get_course_blocks(student, subsection_usage_key)
+        subsection_grade_factory = SubsectionGradeFactory(student, None, subsection_structure)
```

Comment:
Nit: instead of relying on placement of parameters, use keyword arguments for optional parameters:
```
subsection_grade_factory = SubsectionGradeFactory(student, course_structure=subsection_structure)
```

Reply:
Thanks for that. Relatively new at python and wasn't aware you could do keyboard args for optional parameters. Fixed